### PR TITLE
Update sandbox component 'summarizer' to be compatible with latest opennlp-tools release

### PR DESCRIPTION
**Pull request by mawiesne**
- adjusts opennlp-tools to 2.1.0
- introduces Maven pom.xml file and adjusts parent project (org.apache.apache) to version 18
- adjusts Java language level to 11
- adapts code to comply with recent API of `POSTaggerME` from OpenNLP tools
- corrects indentation in many source files to comply with "2 spaces" style
- adds necessary resources required to run existing tests, via maven-download-plugin
- revives JUnit test to actually execute
- adjusts some code to be more modern style

### CODE REVIEW

**Comment by mawiesne** [comment]
@kinow  & @rzo1 - Please check `src/test/resources/wordnet` 
Sadly, all **WordNet** _dict_ files are required during execution of tests. If not present, some tests will fail. Local folders are no option, it seems to me, when we strive for CI / GH actions.

I'm interested to have your opinion / feedback for this resource.

**Comment by mawiesne** [comment]
Idea:

One could carve out those **WordNet** _dict_-dependent tests into separate `integration-tests` via maven-failsafe-plugin, adding those to a separate maven profile.

**Comment by rzo1** [comment]
> @kinow & @rzo1 - Please check `src/test/resources/wordnet` Sadly, all **WordNet** _dict_ files are required during execution of tests. If not present, some tests will fail. Local folders are no option, it seems to me, when we strive for CI / GH actions.
> 
> I'm interested to have your opinion / feedback for this resource.

Do you mean because of the size of Wordnet, its presence in git or its license? 

We could  download and unpack the dict from https://wordnetcode.princeton.edu/wn3.1.dict.tar.gz via Maven Download Plugin in a `pre-integration` test phase. This would of course introduce a dependency on an online resource, but we wouldn`t have the stuff in git. If it's unavailable, the module is most likely of no use (as it seems to require wordnet, no?)

**Comment by kinow** [comment]
Hi @mawiesne 

>@kinow & @rzo1 - Please check src/test/resources/wordnet
>Sadly, all WordNet dict files are required during execution of tests. If not present, some tests will fail. Local folders are no option, it seems to me, when we strive for CI / GH actions.
>
>I'm interested to have your opinion / feedback for this resource.

Just checked, WordNet is 11MB compressed, and ~40MB decompressed :thinking:

For the past 3 months I've been jumping between hotels & airbnb's in Barcelona, and I can tell I got both excellent and really slow Internet reception from mobile&wifi, and running Docker and Maven was not a good experience for some projects. Given we have multiple components in `opennlp-sandbox`, if we start to get many of these 10+ MB files it'd be easy to reach close to 100MB of files for tests, I think.

I think @rzo1 's suggestion of pulling that file when running the tests is probably the best option.

>Do you mean because of the size of Wordnet, its presence in git or its license?

Not sure if the WordNet/Princeton license is compatible with ASF. I had a look and it looks OK to me, but IANAL and haven't had to use these two licenses before - https://wordnet.princeton.edu/license-and-commercial-use. Still, probably confirming it too.

Cheers

File: summarizer/src/main/java/opennlp/summarization/DocProcessor.java [hunk 1]
```diff
@@ -21,17 +21,27 @@
...
-/*
+/**
  * A document processor abstracts a lot of the underlying complexities of parsing the document and 
- * preparing it (e.g. stemming, stop word removal) from the summarization algorithm. The current package
- * supports sentence extraction based algorithms. Thus extracting Sentences from the text is the
- * first step and the basis for the algorithms.
+ * preparing it (e.g. stemming, stop word removal) from the summarization algorithm.
+ * <p>
+ * The current package supports sentence extraction based algorithms.
+ * Thus extracting Sentences from the text is the first step and the basis for the algorithms.
```
Comment (rzo1) [inline summarizer/src/main/java/opennlp/summarization/DocProcessor.java:29]: `sentences`?

File: summarizer/src/main/java/opennlp/summarization/Sentence.java [hunk 2]
```diff
@@ -133,36 +133,36 @@ public int getWordCnt()
...
 		int wrdStrt = 0;
-		StringBuffer b = new StringBuffer();
+		StringBuilder b = new StringBuilder();
 		wrdItr.setText(stringVal);	
 		for(int wrdEnd = wrdItr.next(); wrdEnd != BreakIterator.DONE; 
 				wrdStrt = wrdEnd, wrdEnd = wrdItr.next())
 		{
 			String word = this.getStringVal().substring(wrdStrt, wrdEnd);//words[i].trim();
-			word.replaceAll("\"|'","");
+			word = word.replaceAll("\"|'","");
```
Comment (rzo1) [inline summarizer/src/main/java/opennlp/summarization/Sentence.java:157]: `replace` ?

**Review by rzo1 — APPROVED** [review]

**Comment by rzo1** [comment]
After this PR is merged, I will going to do a JUnit 4 -> JUnit 5 bulk migration, so @mawiesne can use the JUnit 5 features to fight the toggling tests.

File: summarizer/src/main/java/opennlp/summarization/DocProcessor.java (hunk 1 above)
Comment (mawiesne) [reply summarizer/src/main/java/opennlp/summarization/DocProcessor.java:29]: I beautify this in ~2-3 hours.

File: summarizer/src/main/java/opennlp/summarization/Sentence.java (hunk 2 above)
Comment (mawiesne) [reply summarizer/src/main/java/opennlp/summarization/Sentence.java:157]: I beautify this in ~2-3 hours.

**Pull request merged**